### PR TITLE
SMB MW generator hardware

### PR DESCRIPTION
**Pull request by alrik-durand**
## Description
I've edited mw_source_smbv to support other models, espcecially the SMB 100A.
The default parameters are setted so that there is no breaking change.

## Motivation and Context
The constrained were fixed in code. I've changed that to config options.
Now this hardware module can be used with any model similar to the SMBV.

## How Has This Been Tested?
This was tested with a SMB 100 A over an Ethernet connection .

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] I have documented my changes in the changelog (`documentation/changelog.md`)
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have checked that the change does not contain obvious errors (syntax, indentation, mutable default values).
- [ ] I have tested my changes using 'Load all modules' on the default dummy configuration with my changes included.
- [ ] All changed Jupyter notebooks have been stripped of their output cells.

### CODE REVIEW

File: hardware/microwave/mw_source_smbv.py [hunk 1]
```diff
@@ -41,46 +41,56 @@ class MicrowaveSmbv(Base, MicrowaveInterface):
...
     _modtype = 'hardware'
-    _gpib_address = ConfigOption('gpib_address', missing='error')
-    _gpib_timeout = ConfigOption('gpib_timeout', 10, missing='warn')
+
+    # visa address of the hardware : this can be over ethernet, the name is here for
+    # backward compatibility
+    _address = ConfigOption('gpib_address', missing='error')
+    _timeout = ConfigOption('gpib_timeout', 10, missing='warn')
+
+    _configured_hardware = ConfigOption('configured_hardware', 'SMBV100A', missing='warn')
```
Comment (drogenlied) [inline hardware/microwave/mw_source_smbv.py]: Please use the answer of *IDN? to find out what hardware is connected.

File: hardware/microwave/mw_source_smbv.py [hunk 2]
```diff
@@ -92,21 +102,21 @@ def get_limits(self):
         limits = MicrowaveLimits()
         limits.supported_modes = (MicrowaveMode.CW, MicrowaveMode.SWEEP)
 
-        limits.min_power = -145
```
Comment (drogenlied) [inline hardware/microwave/mw_source_smbv.py]: Please leave the absolute minimum/maximum here.
See `hardware/microwave/mw_source_smiq.py` from the master branch about how to do configurable limits correctly.

File: hardware/microwave/mw_source_smbv.py [hunk 3]
```diff
@@ -92,21 +102,21 @@ def get_limits(self):
...
-        limits.list_maxstep = 5.991e9
+        limits.list_maxstep = self._limits_max_frequency - self._limits_min_frequency
         limits.list_maxentries = 1
 
         limits.sweep_minstep = 0.1
-        limits.sweep_maxstep = 5.991e9
+        limits.sweep_maxstep = self._limits_max_frequency - self._limits_min_frequency
         limits.sweep_maxentries = 10001
 
-        if self.model != 'SMBV100A':
```
Comment (drogenlied) [inline hardware/microwave/mw_source_smbv.py]: Keep the old style here, limits should be present for every model where they can be determined.

**Comment by alrik-durand** [comment]
I've tried doing the way you suggested, this code works on my setup !

**Pull request merged**